src/Commands/RunJobs.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace ActiveCollab\JobQueue\Commands;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use ActiveCollab\JobsQueue\Jobs\Job;
+use ActiveCollab\JobsQueue\Queue\QueueInterface;
+/**
+ * @package ActiveCollab\JobQueue\Commands
+ */
+class RunJobs extends Command
+{
+    /**
+     * Configure command
+     */
+    protected function configure()
+    {
+        parent::configure();
+        $this->setName('run_jobs')
+             ->addOption('seconds', 's', InputOption::VALUE_REQUIRED, 'Run jobs for -s seconds before quitting the process', 50)
+             ->addOption('channels', 'k', InputOption::VALUE_REQUIRED, 'Select one or more channels for jobs for process',QueueInterface::MAIN_CHANNEL)
+             ->setDescription('Run jobs that are next in line for up to N seconds');
+    }
+
+    /**
+     * @param  InputInterface  $input
+     * @param  OutputInterface $output
+     * @return int
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $log = $this->log($input);
+
+        // ---------------------------------------------------
+        //  Prepare dispatcher and success and error logs
+        // ---------------------------------------------------
+
+        $dispatcher = $this->getDispatcher($input);
+
+        $jobs_ran = $jobs_failed = [];
+
+        $dispatcher->getQueue()->onJobFailure(function (Job $job) use (&$jobs_failed) {
+            $job_id = $job->getQueueId();
+
+            if (!in_array($job_id, $jobs_failed)) {
+                $jobs_failed[] = $job_id;
+            }
+        });
+
+        // ---------------------------------------------------
+        //  Set max execution time for the jobs in queue
+        // ---------------------------------------------------
+
+        $max_execution_time = (integer)$input->getOption('seconds');
+
+        $output->writeln("There are " . $dispatcher->getQueue()->count() . " jobs in the queue. Preparing to work for {$max_execution_time} seconds.");
+
+        $work_until = time() + $max_execution_time; // Assume that we spent 1 second bootstrapping the command
+        // ---------------------------------------------------
+        //  Set channels for the jobs in queue
+        // ---------------------------------------------------
+        $channels_string = $input->getOption('channels');
```

Comment:
Switch for channel values:

1. Default is to use main
2. If specified, explode and use only these channels - `--channels=mail,app,app1`
3. If `*` use empty array (any channel) - `--channels=*`
4. If trimmed is an empty string, break - `--channels='      '`